code/modules/reagents/chemistry/holder.dm

```diff
@@ -86,6 +91,10 @@
 		var/datum/reagent/R = reagent
 		qdel(R)
 	reagent_list = null
+	if(is_reacting) //If false, reaction list should be cleaned up
+		for(var/reaction in reaction_list)
+			var/datum/equilibrium/E = reaction
+			qdel(E)
```

Comment:
Do the equilibrium datums remove themselves from this list? Else the reference will cause a hard deletion.
Also, you can use the macro for this:
```suggestion
		QDEL_LIST(reaction_list)
```

Reply:
I reworked this to ensure the reactions are properly shut down if it is reacting and destroyed.